docs/hardware.md

```diff
@@ -0,0 +1,98 @@
+include:
+
+- An interferometer
+- A stepper motor for orienting the interferometer's mirror
+- Temperature controllers for getting and setting the temperature of the hot and cold
+  black bodies
+- A separate temperature monitor with sensors recording from various angles around the
+  mirror
+
+All the code for interfacing with the hardware lives in the
+[`finesse.hardware`](../reference/finesse/hardware) module.
+
+## Plugin architecture
+
+Code in the `finesse.hardware` module is not imported into the frontend code
+([`finesse.gui`](../reference/finesse/gui)) directly. Instead, messages are passed back
+and forth using the [PyPubSub](https://pypi.org/project/PyPubSub/) package.
+
+As we want the user to be able to select which devices to use at runtime, the FINESSE
+hardware framework is designed to be modular. This is achieved via a plugin system. Each
+device type and device base type (explained below) is represented by a plugin class
+residing somewhere in the
+[`finesse.hardware.plugins`](../reference/finesse/hardware/plugins) module. To add a new
+plugin, it is sufficient just to define this class in a `.py` file and put it somewhere
+in the plugins directory hierarchy.
+
+### Creating a new device type
+
+A device base type is a class providing a common interface for similar device types
+(e.g. a stepper motor). Each device base class must inherit from `Device` and each
+device class must inherit from a device base class.
+
+You can create a new device base like so:
+
+```py
+class MyBaseType(
+    Device, is_base_type=True, name="my_base_type", description="Example base type"
+):
+    # ...
+```
+
+The `is_base_type=True` is required to register the class as a device base type. `name`
+is the short name for the base type and is used in the topic for PyPubSub messages (see
+more below). `description` provides a human-readable name for the base type, which will
+be displayed in the GUI. It is additionally possible to provide a list of possible names
+for instances of the device, but this is currently only used for temperature controllers
+(to distinguish between the hot and cold black body controllers).
+
+You can create a concrete implementation of `MyBaseType` like so:
+
+```py
+class MyDevice(MyBaseType, description="An example device"):
+    # ...
+```
+
+Certain device base types require extra parameters. For example, to create a USB serial
+device type which also inherits from `MyBaseType`, you can do:
+
+```py
+class MySerialDevice(
+    SerialDevice, MyBaseType, description="An example device", default_baudrate=9600
+):
+    # ...
+```
+
+### Device parameters
+
+Devices can also require additional parameters to be specified by the user, such as port
+and baudrate for serial devices. Details about these parameters are set in an
+`__init_subclass__()` method for the class. These parameters will be passed as arguments
+to the device type's constructor when it is opened.
```

Comment:
This is very unclear - or at least ambiguous. Are these parameters passed to the `__init_subclass__` method, and therefore need to be included in the subclass definition, or are they pass when constructing the object? And example, as you have done above, will be useful.

Reply:
Done now